diff -rcNB verilator-3.880/include/verilated.cpp verilator-3.880.1/include/verilated.cpp
*** verilator-3.880/include/verilated.cpp	2015-11-28 15:36:06.000000000 -0800
--- verilator-3.880.1/include/verilated.cpp	2016-02-27 14:59:32.824149300 -0800
***************
*** 97,107 ****
  // Random reset -- Only called at init time, so don't inline.
  
  IData VL_RAND32() {
! #if defined(_WIN32) && !defined(__CYGWIN__)
!     // Windows doesn't have lrand48(), although Cygwin does.
      return (rand()<<16) ^ rand();
  #else
      return (lrand48()<<16) ^ lrand48();
  #endif
  }
  
--- 97,116 ----
  // Random reset -- Only called at init time, so don't inline.
  
  IData VL_RAND32() {
! #if defined(_WIN32)
!     // Windows doesn't have lrand48(), although (sometimes) Cygwin does.
      return (rand()<<16) ^ rand();
  #else
+   #if defined(__CYGWIN__)
+     #if defined(__STRICT_ANSI__)
+       // In struct-ANSI, Cygwin doesn't have lrand48
+       return (rand()<<16) ^ rand();
+     #else
+       return (lrand48()<<16) ^ lrand48();
+     #endif
+   #else
      return (lrand48()<<16) ^ lrand48();
+   #endif
  #endif
  }
  
